lib/prop/leaky_bucket_strategy.rb

```diff
@@ -0,0 +1,51 @@
+require 'prop/limiter'
+require 'prop/key'
+
+module Prop
+  class LeakyBucketStrategy
+    class << self
+      def update_bucket(cache_key, interval, leak_rate)
+        bucket = Prop::Limiter.reader.call(cache_key) || { :bucket => 0, :last_updated => 0 }
+        now = Time.now.to_i
+        leak_amount = (now - bucket[:last_updated]) / interval * leak_rate
+
+        bucket[:bucket] = [bucket[:bucket] - leak_amount, 0].max
+        bucket[:last_updated] = now
+
+        Prop::Limiter.writer.call(cache_key, bucket)
```

Comment:
Doesn't that depend on the implementation of the writer block passed to `Prop.write`?